python-optional-arguments/optional_params.py

```diff
@@ -0,0 +1,86 @@
+"""
+Optional arguments in Python — consolidated, runnable examples.
+
+This module collects the tutorial's final, good-practice implementations:
+
+- show_list(shopping_list, include_quantities=True)
+- add_item(item_name, quantity, shopping_list=None)
+- add_items_args(shopping_list, *item_names)
+- add_items_kwargs(shopping_list, **things_to_buy)
+
+Run the module directly to see a short demo.
+"""
+
+
+def show_list(shopping_list, include_quantities=True):
+    print()
+    for item_name, quantity in shopping_list.items():
+        if include_quantities:
+            print(f"{quantity}x {item_name}")
+        else:
+            print(item_name)
+
+
+def add_item(item_name, quantity, shopping_list=None):
+    """Add (or increment) an item in a list using the safe 'None' default."""
+    if shopping_list is None:
+        shopping_list = {}
```

Comment:
🔧 **Technical Recommendation**

The name `shopping_list` may imply Python's `list` data structure, while we actually use the variable to store a Python dictionary. How about renaming this variable to `shopping_items` or `item_quantities` instead?